graylog2-server/src/main/java/org/graylog2/database/MongoCollections.java

```diff
@@ -93,4 +109,11 @@ public <T extends MongoEntity> MongoUtils<T> utils(String collectionName, Class<
     public <T extends MongoEntity> MongoUtils<T> utils(MongoCollection<T> collection) {
         return new MongoUtils<>(collection, objectMapper);
     }
+
+    /**
+     * Provides typical utility methods and mutability checks for ScopedEntity objects
+     */
+    public <T extends ScopedEntity> ScopedEntityMongoUtils<T> scopedEntityUtils(MongoCollection<T> collection) {
```

Comment:
What do you think about adding the `EntityScopeService` as a second parameter to this method? That way, we don't have to inject the service into `MongoCollections` and can keep its dependencies lean.

Reply:
Definitely like that better. Thanks for the suggestion.